File: C++ ad y ad/3_ Orientado a objetos/S01.2_ Corredor/Solucion_ejercicio_corredor/Ejercicio_corredor/corredor.cpp
#include "corredor.h"


Corredor::Corredor(string nombre, float velocidadMax)
{
	this->nombre = nombre;
	this->velocidadMax = velocidadMax;

	tieneEnergia = true;
}
Corredor::~Corredor()
{
	cout << "El corredor " << nombre << " fue liberado de la memoria.\n\n";
}


void Corredor::correr()
{
	if (tieneEnergia)
	{
		tieneEnergia = false;

		cout << nombre << " corrio con todas sus fuerzas, alcanzando una velocidad de " << velocidadMax << ".\n";
	}
	else
	{
		cout << nombre << " no tiene energia.\n";
	}
}
void Corredor::descansar()
{
	if (!tieneEnergia)
	{
		tieneEnergia = true;

		cout << nombre << " ha descansado!\n";
	}
	else
	{
		cout << nombre << " ya cuenta con energia!\n";
	}
}

File: C# ad y ad/2_ Orientado a objetos/S01.2_ Corredor/Solucion_ejercicio_corredor/Ejercicio_corredor/Corredor.cs
using System;


namespace Ejercicio_corredor
{
	class Corredor
	{
		string nombre;
		float velocidadMax;
		bool tieneEnergia;


		public Corredor(string nombre, float velocidadMax)
		{
			this.nombre = nombre;
			this.velocidadMax = velocidadMax;

			tieneEnergia = true;
		}


		public void Correr()
		{
			if (tieneEnergia)
			{
				tieneEnergia = false;

				Console.WriteLine(nombre + " corrió con todas sus fuerzas, alcanzando una velocidad de " + velocidadMax + ".");
			}
			else
			{
				Console.WriteLine(nombre + " no tiene energía.");
			}
		}
		public void Descansar()
		{
			if (!tieneEnergia)
			{
				tieneEnergia = true;

				Console.WriteLine("¡" + nombre + " ha descansado!");
			}
			else
			{
				Console.WriteLine("¡" + nombre + " ya cuenta con energía!");
			}
		}
	}
}
